ui/js/models/perf/alerts.js

```diff
@@ -151,19 +151,34 @@ treeherder.factory('PhAlerts', [
             return _.find(phAlertSummaryStatusMap, { id: this.status }).text;
         };
 
+        function _getAlertSummary(id) {
+            // get a specific alert summary
+            var httpTimeout = $q.defer();
```

Comment:
Ah, I see... yes, this is necessary. Could you rename this to something like canceller though, as given in this example http://odetocode.com/blogs/scott/archive/2014/04/24/canceling-http-requests-in-angularjs.aspx ? httpTimeout doesn't seem right.